update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+    elif len(vo_details['voms_endpoints']) == 1:
+        force_voms_admin = True
+    for endpoint in vo_details['voms_endpoints']:
+        template.write("    dict(\n")
+        template.write("        'name', '%s',\n" % endpoint.hostname)
+        template.write("        'host', '%s',\n" % endpoint.hostname)
+        template.write("        'port', %s,\n" % endpoint.vomses_port)
+        template.write("        'adminport', %s,\n" % endpoint.https_port)
+        if not endpoint.is_admin:
+            if not force_voms_admin:
+                template.write("        'type', list('voms-only'),\n")
+            else:
+                sys.stdout.write("WARNING: voms-admin enabled on " +
+                                 "%s as this is the " % endpoint.hostname +
+                                 "only VOMS server (VO card " +
+                                 "inconsistency)\n")
+        template.write("    ),\n")
+    template.write(");\n\n")
+
+    # VOMS Mapping
+    template.write("'voms_mappings' ?= list(\n")
+    legacy_suffix = []
+    for mapping in vo_details['voms_mapping']:
+        suffix = get_legacy_suffix(vo_details['id'], mapping, legacy_suffix)
+        legacy_suffix.append(suffix)
+        suffix2 = get_account_suffix(vo_details['name'], mapping)
+        template.write("    dict(\n")
+        template.write("        'description', '%s',\n" % mapping['desc'])
+        template.write("        'fqan', '%s',\n" % mapping['expr'])
+        template.write("        'suffix', '%s',\n" % suffix)
+        template.write("        'suffix2', '%s',\n" % suffix2)
+        template.write("    ),\n")
+
+    template.write(");\n\n")
+
+    # Base UID
+    base_uid = vo_details['id'] * 1000
+    template.write("'base_uid' ?= %i;\n" % base_uid)
+    template.close()
+
+
+def update_voms_servers(vo_details, basedir, namespaces, debug=False):
+    """write template containing the voms certficate
+    """
+    voms_servers = {}
+    for vo_config in vo_details:
+        if not vo_config['voms_endpoints']:
+            sys.stderr.write("No VOMS server defined in any VO\n")
+        else:
+            for server in vo_config['voms_endpoints']:
+                if server.hostname not in voms_servers:
+                    voms_servers[server.hostname] = server.cert
+    for (hostname, cert) in voms_servers.items():
+        server_cert_ns = namespaces['cert_dir'] + os.path.sep + hostname
+        server_cert_tpl = basedir + os.path.sep + server_cert_ns + ".pan"
+        if cert is None:
+            sys.stdout.write("No certicate defined for %s " % hostname +
+                             "voms server.\n")
+            x509_cert = None
+        else:
+            try:
+                x509_cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert)
+
+            except crypto.Error as e:
+                sys.stderr.write("ERROR Certificate for %s " % hostname +
+                                 "could not be loaded: %s\n" % str(e))
```

Comment:
No need to double stringify, `%s` converts to string, so the `str()` around `e` is redundant, also you can pass tuples and lists to `%`.

```python
sys.stderr.write("ERROR Certificate for %s could not be loaded: %s\n" % (hostname, e))
```

Reply:
I have split in two lines to avoid pep8 errors.

Reply:
The str call has been removed in commit 091e53d42df1e799fe46897c9ff4c91567124ada.